### PR TITLE
PTEUDO-1247 Refactor per-request parameters to be injected into context

### PR DESCRIPTION
**Pull request by leandrorichardtoledo**
- It addresses the ticket: [PTEUDO-1247](https://infoblox.atlassian.net/browse/PTEUDO-1247);
- Removed input and mode global vars from the reconciler struct;
- Refactored code for passing the request info and mode when needed;

### CODE REVIEW

**Review by EvertonCalgarotto — APPROVED** [review]
Looks good!

File: pkg/databaseclaim/databaseclaim.go [hunk 1]
```diff
@@ -1355,73 +1263,70 @@ func (r *DatabaseClaimReconciler) readResourceSecret(ctx context.Context, secret
 
 // getDynamicHostName is used to name the crossplane
 // dbinstance CRs
-func (r *DatabaseClaimReconciler) getDynamicHostName(dbClaim *v1.DatabaseClaim) string {
+func (r *DatabaseClaimReconciler) getDynamicHostName(hash string, dbClaim *v1.DatabaseClaim) string {
```
Comment (drewwells) [inline pkg/databaseclaim/databaseclaim.go:1266]: Remove this from the reconciler object. Pass in the 'dbidentifierprefix' which is just the env name. Feel free to just rename it to envname as well.

File: pkg/databaseclaim/requestinfo.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	EnableReplicationRole      bool
+	EnableSuperUser            bool
+	EnablePerfInsight          bool
+	EnableCloudwatchLogsExport []*string
+	BackupRetentionDays        int64
+	CACertificateIdentifier    string
+}
+
+// setRequestInfo load base values and configs to kick off the database claim request.
+func (r *DatabaseClaimReconciler) setRequestInfo(dbClaim *v1.DatabaseClaim) (requestInfo, error) {
```
Comment (drewwells) [inline pkg/databaseclaim/requestinfo.go:27]: The goal is to split all of this code up into smaller pieces. To get there, this should not hang off the reconciler struct and make a bunch of calls into the reconciler to find things. 

Change this to receive the Config and dbClaim as parameters. Made some comments above on getDynamicHostName to remove that dependence on reconciler struct too.
```suggestion
func NewRequestInfo(config *DatabaseClaimConfig, dbClaim *v1.DatabaseClaim) (requestInfo, error) {
```

**Review by drewwells — CHANGES_REQUESTED** [review]
Some unit tests on this would be helpful to verify functionality. Made some comments to remove this from the reconciler struct so it can run independently

File: pkg/databaseclaim/requestinfo.go (hunk 2 above)
Comment (drewwells) [inline pkg/databaseclaim/requestinfo.go:27]: pass in the viper config and things to call `r.getDynamicHostName(hostParams.Hash(), dbClaim)`, so the reconciler method receiver can be removed.
`NewRequestInfo(dbClaim *v1.DatabaseClaim, ViperConfig...)`

File: pkg/databaseclaim/requestinfo.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+		enablePerfInsight       bool
+		cloudwatchLogsExport    []*string
+		backupRetentionDays     int64
+		caCertificateIdentifier string
+	)
+
+	backupRetentionDays = basefun.GetBackupRetentionDays(r.Config.Viper)
+	caCertificateIdentifier = basefun.GetCaCertificateIdentifier(r.Config.Viper)
+	enablePerfInsight = basefun.GetEnablePerfInsight(r.Config.Viper)
+	enableCloudwatchLogsExport := basefun.GetEnableCloudwatchLogsExport(r.Config.Viper)
```
Comment (drewwells) [inline pkg/databaseclaim/requestinfo.go:39]: This is a great reason to stop passing around viperconfig, it makes calling basefun methods very difficult. We should just have a config package to find these configuration parameters, then we could just do
```
backupRetentionDays = config.BackupRetentionDays
caCertificateIdentifier = config.CACertificateIdentifier
... etc
```

**Review by drewwells — APPROVED** [review]

**Pull request merged**